invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/AdminForm.js

```diff
@@ -37,17 +43,38 @@ export class AdminForm extends Component {
     const { apiEndpoint, pid, successCallback, create } = this.props;
     const { addNotification } = this.context;
     let response;
+
+    const transformedValues = mapValues(values, (value, key) => {
+      const fieldSchema = this.props.resourceSchema[key];
+      if (fieldSchema?.metadata?.type === "json") {
+        try {
+          if (value === "") {
+            return null;
+          } else if (typeof value === "object") {
+            return value;
+          } else {
+            return JSON.parse(value);
+          }
+        } catch (e) {
+          console.error(`Error parsing JSON for field ${key}:`, e);
+          actions.setFieldError(key, "Invalid JSON format");
+          throw e;
```

Comment:
minor/shelve: I feel that the whole conversion/validation logic is spread throughout the module, instead of specific components/classes being responsible for it. Not sure if it should be addressed in this PR, but definitely something to look into in the future.

On a separate note, it's likely that in the future we'll look into integrating a dedicated code-editor style of form field for JSON (or YAML), where I would expect the form control to be responsible for all this logic.